src/e2e/java/teammates/e2e/cases/e2e/StudentProfilePageE2ETest.java

```diff
@@ -0,0 +1,112 @@
+        testData.students.get("studentWithExistingProfile").email = student2Email;
+
+        // also inject instructor account for this test
+        String instructorGoogleId = TestProperties.TEST_INSTRUCTOR_ACCOUNT;
+        String instructorEmail = instructorGoogleId + "@gmail.com";
+        testData.accounts.get("SProfileUiT.instr").googleId = instructorGoogleId;
+        testData.accounts.get("SProfileUiT.instr").email = instructorEmail;
+        testData.instructors.get("SProfileUiT.instr.CS2104").googleId = instructorGoogleId;
+        testData.instructors.get("SProfileUiT.instr.CS2104").email = instructorEmail;
+        testData.instructors.get("SProfileUiT.instr.CS2103").googleId = instructorGoogleId;
+        testData.instructors.get("SProfileUiT.instr.CS2103").email = instructorEmail;
+
+        removeAndRestoreDataBundle(testData);
+    }
+
+    @Test
+    public void allTests() throws Exception {
+        // Do not change the order
+        testNavLinkToPage();
+        testContent();
+    }
+
+    private void testNavLinkToPage() {
+        StudentHomePage shp = getHomePage().clickStudentLogin().loginAsStudent(
+                TestProperties.TEST_STUDENT2_ACCOUNT, TestProperties.TEST_STUDENT2_PASSWORD);
+        profilePage = shp.loadProfileTab();
+    }
+
+    private void testContent() throws Exception {
+        // assumes it is run after NavLinks Test
+        // (ie already logged in as studentWithExistingProfile
+        ______TS("Typical case: Log in with filled profile values");
+
+        profilePage.ensureProfileContains("Ben", "i.m.benny@gmail.tmt", "TEAMMATES Test Institute 4",
+                "Singaporean", StudentProfileAttributes.Gender.MALE, "I am just another student :P");
+
+        ______TS("Typical case: no picture");
+        profilePage.editProfileThroughUi("short.name", "e@email.tmt", "inst", "American",
+                StudentProfileAttributes.Gender.FEMALE, "this is enough!$%&*</>");
+
+        profilePage.ensureProfileContains("short.name", "e@email.tmt", "inst", "American",
+                StudentProfileAttributes.Gender.FEMALE, "this is enough!$%&*</>");
+        profilePage.verifyStatusMessage(Const.StatusMessages.STUDENT_PROFILE_EDITED);
+
+        ______TS("Typical case: picture upload and edit");
+
+        profilePage.fillProfilePic("src/test/resources/images/profile_pic.png");
+        profilePage.uploadPicture();
+        profilePage.verifyStatusMessage(Const.StatusMessages.STUDENT_PROFILE_PICTURE_SAVED);
+
+        profilePage.showPictureEditor();
+        profilePage.waitForUploadEditModalVisible();
+
+        profilePage.editProfilePhoto();
+        profilePage.ensureProfileContains("short.name", "e@email.tmt", "inst", "American",
+                StudentProfileAttributes.Gender.FEMALE, "this is enough!$%&*</>");
+        profilePage.verifyPhotoSize("295px", "295px");
+
+        JSONObject studentProfileAttributes = new JSONObject(
```

Comment:
Related to my comment in the `BackDoor` class: This shouldn't be a `JSONObject`. It should be the proper `StudentProfileAttributes` object.